lib/executor/executor.cpp

```diff
@@ -206,5 +207,190 @@ Executor::asyncInvoke(const Runtime::Instance::FunctionInstance *FuncInst,
+    File.close();
+  } else {
+    // TODO change to assuming
+    //  Handle error opening file
+    throw std::ios_base::failure("Failed");
+  }
+}
+
+AST::CustomSection Executor::collectProcessInformation() {
+  spdlog::info("Constructing Core");
+  // AST node
+  AST::CustomSection Core;
+  Core.setName("core");
+
+  // Trying to insert this in leads to wasmgdb
+  // saying it is an unsupported process-type
+  // std::string Name = "a.wasm";
+  // std::vector<Byte> Data(Name.begin(), Name.end());
+  // if (Data.size() != 0) { // case where the module has an empty name field
+  //   Content.insert(Content.end(), Data.begin(), Data.end());
+  // } else {
+  //   Content.insert(Content.begin(), {0x00, 0x00});
+  // }
+  auto &Content = Core.getContent();
+  Content.insert(Content.begin(), {0x00, 0x00});
+
+  return Core;
+}
+
+AST::MemorySection Executor::collectMemory(Runtime::StackManager &StackMgr) {
+  spdlog::info("Collecting memory section");
+
+  AST::MemorySection MemSec;
+  const auto *CurrentInstance = StackMgr.getModule();
+  auto MemInstances = CurrentInstance->getMemoryInstances();
+  std::cout << MemInstances.size() << "\n";
+  std::cout << MemInstances[0]->getMemoryType().getLimit().hasMax() << "\n";
+  std::cout << MemInstances[0]->getMemoryType().getLimit().getMax() << "\n";
+  std::cout << MemInstances[0]->getMemoryType().getLimit().getMin() << "\n";
+  MemInstances[0]->getMemoryType().getLimit();
+  // In the current version of WebAssembly, at most one memory may be defined
+  // or imported in a single module, and all constructs implicitly reference
+  // this memory.
+  //
+  // If hasMax, set Max to that.
+  // Else set whatever Min exists
+
+  return MemSec;
+}
+
+// Data section
+// TODO malformed section
+AST::DataSection Executor::collectDataSection(Runtime::StackManager &StackMgr) {
+  spdlog::info("Collecting Memories");
+
+  AST::DataSection Datasec;
+
+  const auto *CurrentInstance = StackMgr.getModule();
+  auto &DataInstances = CurrentInstance->getOwnedDataInstances();
+
+  AST::DataSegment Seg;
+  auto &Content = Seg.getData();
+
+  for (auto &Iter : DataInstances) {
+    Content.insert(Content.end(), Iter->getData().begin(),
+                   Iter->getData().end());
+  }
+  if (Content.size() == 0) {
+    std::cout << "empty\n";
+    std::cout << Content.max_size();
+    Content.insert(Seg.getData().end(), 0x00);
+  }
+  std::cout << Content.size();
+  Datasec.getContent().push_back(Seg);
+  return Datasec;
+}
+
+// corestack   ::= customsec(thread-info vec(frame))
+// thread-info ::= 0x0 thread-name:name
+// frame       ::= 0x0 funcidx:u32 codeoffset:u32
+//                     locals:vec(value) stack:vec(value)
+AST::CustomSection Executor::collectCoreStack(Runtime::StackManager &StackMgr) {
+  spdlog::info("Constructing CoreStack");
+
+  AST::CustomSection CoreStack;
+  CoreStack.setName("corestack");
+
+  auto Frames = StackMgr.getAllFrames();
+  std::reverse(Frames.begin(), Frames.end());
```

Comment:
You should never apply reversing to all frames. Use `rbegin()` and `rend()` instead.